### DISCUSSION TITLE
How to setup own cdn with blocks?

### DISCUSSION THREAD

**Discussion by levleontiev** [lowdefy/lowdefy#1507] [Questions and Help]
All the links here in comments are dead, the renderer package is removed from repo. 

How should one create self-hosted server with blocks?

**Comment by Gervwyk** [comment]
Hi @levleontiev 
This repo has a v3 branch where everything for v3 is available: https://github.com/lowdefy/lowdefy/tree/v3 so you should be able to just check the links on v3 branch.

With that said, v4 is in rc.5 and stable, we are just wrapping up the last bit of docs and website changes. I would recommend converting to v4, and converting your blocks to plugins. It will be a much better experience as all your plugins are bundled with your app. 

See the current v4 migration guide and v4 docs here: https://docs-git-develop-lowdefy.vercel.app/v3-to-v4

**Reply by gramian** [reply]
Could you be more specific with regard to the links in the v3 branch? The commits ahead don't seem to hint at anything related to a standalone CDN.

Thank you

**Reply by Gervwyk** [reply]
Hi @gramian 
So the v3 docs are hosted at: https://docs-v3.lowdefy.com

So the /custom-blocks page will be.
https://docs-v3.lowdefy.com/custom-blocks

We are still hosting the v3 docs, but the backlinks in the docs might link to a `https://docs.lowdefy.com/custom-blocks` so you'd need to change that to `https://docs-v3.lowdefy.com/custom-blocks` instead if you hit a 404. But for the most part, these links should be working fine. Is there any specific link that is not working?

**Reply by gramian** [reply]
Sorry, I think I misunderstood question and answer. I was looking for info on how to avoid remote server calls for blocks.

**Reply by Gervwyk** [reply]
Ok cool. Yeah, in v3 blocks are served separately from the main build bundle using a micro-frontend architecture, with webpack module federation. We simplified this in v4 by having a single build bundle, which includes all block code, so no more loading blocks individually.
